joomla/joomla-cms
# [4.0] SVG favicons

Pull Request for Issue # .

### Summary of Changes
There are 2 things done in this PR:
- The code for autogenerating the favicon link is moved to the renderer
- Both templates moved to using SVG favicons

##### The code for autogenerating the favicon link is moved to the renderer
Why?
The code used to live in the Document function `_loadTemplate` and would always ran when parsing the template index (or whatever was instructed to use as index). The proposed change here moves the responsibility to the MetasRenderer because the code spits out a head link and all other headlinks are handled by this renderer. Also the code will never run if a favicon has already been set, meaning it's more efficient (and we like efficiency, right?)!

##### Both templates moved to using SVG favicons
Why?
Sort answer: because it's 2020. Longer version:
According to https://caniuse.com/link-icon-svg 89.21% of the Browsers already support SVG favicons. Also there's a fallback so in our case it's always 100%.
SVG icons are vector based, meaning they will always look sharp!
SVG icons support light/dark themes
Usually they're lighter than the .ico

### Testing Instructions
- Check the browser tab icon for both the cassiopeia and Atum templates.
- Open `templates/cassiopeia/index.php` and delete or mute these lines (40-44)
```php
// Browsers support SVG favicons
$this->addHeadLink(HTMLHelper::image('joomla-favicon.svg', '', [], true, 1),'icon', 'rel', ['type' => 'image/svg+xml']);
$this->addHeadLink(HTMLHelper::image('favicon.ico', '', [], true, 1), 'alternate icon', 'rel', ['type' => 'image/vnd.microsoft.icon']);
$this->addHeadLink(HTMLHelper::image('joomla-favicon-pinned.svg', '', [], true, 1),'mask-icon', 'rel', ['color' => '#000']);
```
- copy the file `templates/cassiopeia/images/favicon.ico` to `templates/cassiopeia/favicon.ico` and check that you get an icon in the front-page

### Actual result BEFORE applying this Pull Request



### Expected result AFTER applying this Pull Request
Dark Mode
![Screenshot 2020-11-20 at 14 30 46](https://user-images.githubusercontent.com/3889375/99805694-3230f200-2b3d-11eb-988e-e9944b050b5b.png)
Light Mode
![Screenshot 2020-11-20 at 14 31 44](https://user-images.githubusercontent.com/3889375/99805753-42e16800-2b3d-11eb-9500-d6448911498d.png)
Safari Pin tab icon
![Screenshot 2020-11-20 at 14 30 30](https://user-images.githubusercontent.com/3889375/99805791-4e349380-2b3d-11eb-9bbf-b552a4efe884.png)


### Documentation Changes Required
IDK but there are some great articles out there for the svg favicons, eg https://css-tricks.com/svg-favicons-and-all-the-fun-things-we-can-do-with-them/
